src/internal_events/eventstoredb_metrics.rs

```diff
@@ -25,27 +64,80 @@ pub struct EventStoreDbStatsParsingError {
 
 impl InternalEvent for EventStoreDbStatsParsingError {
     fn emit_logs(&self) {
-        error!(message = "JSON parsing error.", error = ?self.error);
+        error!(
+            message = "JSON parsing error.",
+            error = ?self.error,
+            error_type = "parse_failed",
+            stage = "processing",
+        );
     }
 
     fn emit_metrics(&self) {
+        counter!(
+            "component_errors_total", 1,
+            "stage" => "processing",
+            "error_type" => "parse_failed",
+        );
+        // deprecated
         counter!("parse_errors_total", 1);
     }
 }
 
-pub struct EventStoreDbMetricsReceived {
-    pub events: usize,
+pub struct EventStoreDbMetricsBytesReceived {
     pub byte_size: usize,
 }
 
-impl InternalEvent for EventStoreDbMetricsReceived {
+impl InternalEvent for EventStoreDbMetricsBytesReceived {
     fn emit_logs(&self) {
-        debug!("Stats scraped.");
+        trace!(
+            message = "Bytes received.",
+            byte_size = %self.byte_size,
+            protocol = "http",
+        );
     }
 
     fn emit_metrics(&self) {
-        counter!("component_received_events_total", self.events as u64);
-        counter!("events_in_total", self.events as u64);
+        counter!(
+            "component_received_bytes_total", self.byte_size as u64,
+            "protocol" => "http",
+        );
+    }
+}
+
+pub struct EventStoreDbMetricsEventsReceived {
+    pub count: usize,
+    pub byte_size: usize,
+}
+
+impl InternalEvent for EventStoreDbMetricsEventsReceived {
+    fn emit_logs(&self) {
+        trace!(message = "Events received.", count = %self.count, byte_size = %self.byte_size);
+    }
+
+    fn emit_metrics(&self) {
+        counter!("component_received_events_total", self.count as u64);
+        counter!(
+            "component_received_event_bytes_total",
+            self.byte_size as u64
+        );
+        // deprecated
+        counter!("events_in_total", self.count as u64);
         counter!("processed_bytes_total", self.byte_size as u64);
     }
 }
```

Comment:
I think this can reuse the common `EventsReceived` in `lib/vector-common/src/internal_events`